playbooks/setup-cluster-env.yml

```diff
@@ -0,0 +1,111 @@
+## Disclaimer:
+# This playbook is not officially supported and comes with no guarantees.
+# Use it at your own risk. Ensure you thoroughly test it in your environment
+# before deploying to production.
+#
+# Playbook: Setup pipeline environment — cluster_name, ocp_primary_nic, and ocp_secondary_nic
+#
+# Description:
+#   This playbook configures pipeline parameters based on the provided OCP version:
+#     - Determines and sets network interface cards (NICs) for the cluster (primary OCP NIC and secondary SR-IOV NIC)
+#     - Selects a cluster name based on the OpenShift Container Platform (OCP) version
+#     - Writes the selected NICs and cluster name to files for later use
+#
+# Usage:
+#   This playbook is intended to run locally on the localhost.
+#
+# Variables:
+#   - dest_directory: Directory where output files will be saved (default: /tmp/)
+#   - ocp_secondary_nic_file_name: Filename for the secondary NIC
+#   - ocp_primary_nic_file_name: Filename for the primary NIC
+#   - cluster_name_file_name: Filename for the cluster name
+#   - cluster_release_map: Mapping of OCP versions to cluster names and NICs
+#
+# Tasks:
+#   - Gather the latest OCP version for the given minor release
+#   - Select a secondary NIC based on the OCP z-stream version (each z-stream uses a unique NIC)
+#   - Select a primary NIC (different from the secondary NIC)
+#   - Set and write the cluster name, primary NIC, and secondary NIC to files
+#
+# Command Example:
+#     ansible-playbook ./playbooks/setup-cluster-env.yml --extra-vars 'release=4.17'
+#
+# Notes:
+#   - This playbook assumes the `release` variable is provided externally before execution.
+#   - The `ocp_version_facts` role must be available for this playbook to function correctly.
+---
+- name: Setup pipeline environment
+  hosts: localhost
+  gather_facts: false
+  vars:
+    dest_directory: /tmp/
+    ocp_secondary_nic_file_name: "secondary_nic"
+    ocp_primary_nic_file_name: "ocp_nic"
+    cluster_name_file_name: "cluster_name"
+    cluster_release_map:
+      "4.17":
+        cluster_name: kni-qe-92
+        nics:
+          - mlx
+          - intel_710
+          - intel_810
+      "4.16":
+        cluster_name: hlx-cl3
+        nics:
+          - mlx
+          - intel_710
+          - intel_810
```

Comment:
if we want to make this playbook reusable - maybe we should consider moving this to env file and not inside the playbook

Reply:
This one is relevant only for single cnf network team. I don't see any use case for other teams for this particular playbook.

Reply:
To be honest, I'm still not sure if we are going to reuse it. We just need something to get started. Let's see if we end up reusing it, and once we have a new use case, we'll make it more generic.

Reply:
/lgtm